File: scala/src/test/scala/com/shekhargulati/ninetynine_problems/_02_arithmetic/P41Test.scala
package com.shekhargulati.ninetynine_problems._02_arithmetic

import org.scalatest.{FunSpec, Matchers}

class P41Test extends FunSpec with Matchers {

  describe("P41 Spec") {
    it("calculate phi of 10090 using P39 solution"){
      val p = P39.totient_phi(10090)
      p should be(4032)
    }

    it("calculate phi of 10090 using P40 solution"){
      val p = P40.phi(10090)
      p should be(4032)
    }
  }

}

File: java8/src/test/java/com/shekhargulati/ninetynine_problems/_02_arithmetic/P41Test.java
package com.shekhargulati.ninetynine_problems._02_arithmetic;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.equalTo;
import static org.junit.Assert.assertThat;

public class P41Test {

    @Test
    public void shouldCalculatePhiOf10090UsingP39() throws Exception {
        long p = P39.phi(10090);
        assertThat(p, equalTo(4032L));
    }

    @Test
    public void shouldCalculatePhiOf10090UsingP40() throws Exception {
        long p = P40.phi(10090);
        assertThat(p, equalTo(4032L));
    }
}